pkg/apis/networking/validation/validation.go

```diff
@@ -778,9 +783,10 @@ func ValidateServiceCIDR(cidrConfig *networking.ServiceCIDR) field.ErrorList {
 	}
 	// validate cidrs are dual stack, one of each IP family
 	if len(cidrConfig.Spec.CIDRs) == 2 {
+		// ignore the error since we validate later
 		isDual, err := netutils.IsDualStackCIDRStrings(cidrConfig.Spec.CIDRs)
-		if err != nil || !isDual {
-			allErrs = append(allErrs, field.Invalid(fieldPath, cidrConfig.Spec, "may specify no more than one IP for each IP family, i.e 192.168.0.0/24 and 2001:db8::/64"))
+		if err == nil && !isDual {
```

Comment:
it's really hard to tell if IsValidCIDR below covers all the things that would return err here, let's leave this as-is for this PR and also return this error if err != nil

Reply:
could move the individual CIDR validation loop above the dualstack validation block to return early if there are invalid CIDRs in the list, while still keeping an error here

Reply:
fixed avoiding the dependency on the helper